js/components/grid/grid.js.jsx

```diff
@@ -8,50 +8,169 @@ class Grid extends React.Component {
     this.addSelectedCourse = this.addSelectedCourse.bind(this);
     this.removeSelectedCourse = this.removeSelectedCourse.bind(this);
     this.clearSelectedCourses = this.clearSelectedCourses.bind(this);
+
     this.addSelectedLecture = this.addSelectedLecture.bind(this);
     this.removeSelectedLecture = this.removeSelectedLecture.bind(this);
+    this.createNewCourse = this.createNewCourse.bind(this);
+    this.createNewLecture = this.createNewLecture.bind(this);
     this.state = {
       selectedLectures: [],
       selectedCourses: []
     };
   }
 
+  // get the previously selected courses and lecture sections from local storage
   componentDidMount() {
-    this.setState({ selectedLectures: generateData() });
+    let selectedCoursesLocalStorage = localStorage.getItem('selectedCourses');
+    let selectedLecturesLocalStorage = localStorage.getItem('selectedLectures');
+
+    if (!selectedLecturesLocalStorage) {
+      selectedLecturesLocalStorage = [];
+    } else {
+      try {
+        this.setState({selectedLectures: JSON.parse(selectedLecturesLocalStorage)});
+      }
+      catch (e) {
+        console.log(e)
+      }
+    }
+
+    if (!selectedCoursesLocalStorage) {
+      selectedCoursesLocalStorage = [];
+    } else {
+      selectedCoursesLocalStorage = selectedCoursesLocalStorage.split('_');
+      selectedCoursesLocalStorage.forEach((courseCode) => {
+        try {
```

Comment:
It's not clear what in `addSelectedCourse` would raise an error, so you should probably remove this.